iothub/device/samples/PnpDeviceSamples/Thermostat/Program.cs

```diff
@@ -27,7 +28,21 @@ public class Program
         // DTDL interface used: https://github.com/Azure/opendigitaltwins-dtdl/blob/master/DTDL/v2/samples/Thermostat.json
         private const string ModelId = "dtmi:com:example:Thermostat;1";
 
+        // This environment variable indicates if DPS or IoT Hub connection string will be used to provision the device.
+        // Expected values: (case-insensitive)
+        // "DPS" - The sample will use DPS to provision the device.
+        // "connectionString" - The sample will use IoT Hub connection string to provision the device.
+        private static readonly string s_deviceSecurityType = Environment.GetEnvironmentVariable("IOTHUB_DEVICE_SECURITY_TYPE");
```

Comment:
Since it is a requirement to have a similar usage experience across all sdks, all languages are using environment variables, and not language specific command line parsers (even though they are definitely more convenient).

Reply:
Who set that requirement?

Reply:
This was set when we first started writing the samples this summer. The idea was that, if someone where to give the python sample a try, and then decide to try the .NET sample, the on-boarding time should be min, and experience should be similar.
We know that java and .NET have similar experience with these command line parsers, I'll sync up with scripting team to see if they have a similar option. If not, we might have to stick to env var for now.